### PR TITLE
Bitmex: implement pair normalization + add other improvements

### PR DESCRIPTION
**Pull request by olibre**
### Description of code - what bug does this fix / what feature does this add?

* Fix https://github.com/bmoscon/cryptofeed/issues/361
* Add some type hitting
* Add try/except to see the received message from exchange

- [ ] - Tested
- [ ] - Changelog updated
- [ ] - Tests run and pass
- [ ] - Flake8 run and all errors/warnings resolved

### CODE REVIEW

File: cryptofeed/pairs.py [hunk 1]
```diff
@@ -40,46 +41,60 @@ def gen_pairs(exchange: str, key_id=None):
...
+            del norm[k]
     ret = {}
-    r = requests.get('https://api.bitfinex.com/v2/tickers?symbols=ALL').json()
-    for data in r:
-        pair = data[0]
+    for pair in [t[0] for t in tickers]:
         if pair[0] == 'f':
-            continue
-        normalized = pair[1:-3] + PAIR_SEP + pair[-3:]
-        normalized = normalized.replace('UST', 'USDT')
```
Comment (yohplala) [inline cryptofeed/pairs.py:77]: Hi @olibre 
Please, keep this normalization step. Bitfinex knows `USDT` as `UST`.

Also, could you add a line making the same to replace `BCHN`as `BCH`?
Most of the main other exchanges have kept `BCH` for Bitcoin Cash Node (Kraken, Huobi & Binance most notably)

Thanks

**Review by bmoscon — APPROVED** [review]

**Review by bmoscon — COMMENTED** [review]
looks good thanks!

File: cryptofeed/pairs.py [hunk 2]
```diff
@@ -250,62 +265,78 @@ def okcoin_pairs():
...
-def gateio_pairs():
+def gateio_pairs() -> Dict[str, str]:
     r = requests.get("https://api.gateio.ws/api/v4/spot/currency_pairs").json()
     return {data['id'].replace("_", PAIR_SEP): data['id'] for data in r}
 
 
-def bitmex_pairs():
+def bitmex_pairs() -> Dict[str, str]:
     r = requests.get("https://www.bitmex.com/api/v1/instrument/active").json()
     return {entry['symbol']: entry['symbol'] for entry in r}
```
Comment (yohplala) [inline cryptofeed/pairs.py:336]: hi @olibre 
I am sorry, I don't understand. Nothing has changed here regarding Bitmex pair normalization. You indicate that this PR implements pair normalization for Bitmex. Am I being dumb?

File: cryptofeed/pairs.py (hunk 1 above)
Comment (bmoscon) [reply cryptofeed/pairs.py:77]: it converts UST to USDt. I added a line that makes it USDT. I'll add something to force BCHN -> BCH

Comment (bmoscon) [reply cryptofeed/pairs.py:77]: done

File: cryptofeed/pairs.py (hunk 2 above)
Comment (olibre) [reply cryptofeed/pairs.py:336]: Yes @yohplala you are right. I am confused => I have fixed this PR title. I will fix the Bitmex pair retrieval later...

Comment (olibre) [reply cryptofeed/pairs.py:336]: Yes @yohplala you are right. I am confused => I have fixed this PR title.

**Pull request merged**